frame/nomination-pools/src/lib.rs

```diff
@@ -2224,9 +2224,15 @@ impl<T: Config> Pallet<T> {
 		T::Currency::make_free_balance_be(&bonded_pool.bonded_account(), Zero::zero());
 
 		Self::deposit_event(Event::<T>::Destroyed { pool_id: bonded_pool.id });
+		Self::remove_metadata(bonded_pool.id);
 		bonded_pool.remove();
 	}
 
+	/// Remove bonded pool metadata.
+	pub fn remove_metadata(bonded_pool_id: PoolId) {
+		Metadata::<T>::remove(bonded_pool_id);
+	}
```

Comment:
no need for a 1 liner helper function please :D

Reply:
I think it was cool to kinda suggest it’s importance, but I agree it’s a bit too much boilerplate 😅